src/parsers/proc_stat.cpp

```diff
@@ -0,0 +1,219 @@
+    // clang-format on
+
+    enum token
+    {
+        TOTAL       = 0,
+        MIN_COUNT   = 1,
+    };
+
+    auto tokens = utils::split(value);
+    if (tokens.size() < MIN_COUNT)
+    {
+        throw parser_error(
+            "Corrupted sequence - Unexpected tokens count", value
+        );
+    }
+
+    try
+    {
+        proc_stat::sequence sequence;
+
+        utils::stot(tokens[TOTAL], out.total);
+
+        for (size_t i = MIN_COUNT; i < tokens.size(); i++) {
+            unsigned long long value;
+            utils::stot(tokens[i], value);
+            out.per_item.push_back(value);
+        }
+    }
+    catch (const std::invalid_argument& ex)
+    {
+        throw parser_error("Corrupted sequence - Invalid argument", value);
+    }
+    catch (const std::out_of_range& ex)
+    {
+        throw parser_error("Corrupted sequence - Out of range", value);
+    }
+}
+
+static void to_cpu(const std::string& value, proc_stat::cpu& out)
+{
+    // Some examples:
+    // clang-format off
+    // 21497341 899627 8830588 433191163 93490 0 1844976 0 0 0
+    // 2684811 115236 1094082 54162041 10674 0 890071 0 0 0
+    // clang-format on
+
+    enum token
+    {
+        USER        = 0,
+        NICE        = 1,
+        SYSTEM      = 2,
+        IDLE        = 3,
+        MIN_COUNT   = 4,
+        IOWAIT      = 4,
+        IRQ         = 5,
+        SOFTIRQ     = 6,
+        STEAL       = 7,
+        GUEST       = 8,
+        GUEST_NICE  = 9,
+        COUNT
+    };
+
+    auto tokens = utils::split(value);
+    if (tokens.size() < MIN_COUNT || tokens.size() > COUNT)
+    {
+        throw parser_error("Corrupted cpu - Unexpected tokens count", value);
+    }
+
+    try
+    {
+        utils::stot(tokens[USER], out.user);
+        utils::stot(tokens[NICE], out.nice);
+        utils::stot(tokens[SYSTEM], out.system);
+        utils::stot(tokens[IDLE], out.idle);
+        if (tokens.size() > IOWAIT) {
+            utils::stot(tokens[IOWAIT], out.iowait);
+        }
+        if (tokens.size() > IRQ) {
+            utils::stot(tokens[IRQ], out.irq);
+        }
+        if (tokens.size() > SOFTIRQ) {
+            utils::stot(tokens[SOFTIRQ], out.softirq);
+        }
+        if (tokens.size() > STEAL) {
+            utils::stot(tokens[STEAL], out.steal);
+        }
+        if (tokens.size() > GUEST) {
+            utils::stot(tokens[GUEST], out.guest);
+        }
+        if (tokens.size() > GUEST_NICE) {
+            utils::stot(tokens[GUEST_NICE], out.guest_nice);
+        }
+    }
+    catch (const std::invalid_argument& ex)
+    {
+        throw parser_error("Corrupted cpu - Invalid argument", value);
+    }
+    catch (const std::out_of_range& ex)
+    {
+        throw parser_error("Corrupted cpu - Out of range", value);
+    }
+}
+
+static void parse_cpu(const std::string& value, proc_stat& out)
+{
+    proc_stat::cpu cpu;
+    memset(&cpu, 0, sizeof(cpu));
```

Comment:
A better implementation will assign default values to fields in the `cpu` struct